Mathlib/Order/Hom/Lattice.lean

```diff
@@ -1588,3 +1588,263 @@ theorem symm_dual_comp (g : BoundedLatticeHom βᵒᵈ γᵒᵈ) (f : BoundedLat
 #align bounded_lattice_hom.symm_dual_comp BoundedLatticeHom.symm_dual_comp
 
 end BoundedLatticeHom
+
+/-! ### `with_top`, `with_bot` -/
+
+namespace SupHom
+variable [SemilatticeSup α] [SemilatticeSup β] [SemilatticeSup γ]
+
+/-- Adjoins a `⊤` to the domain and codomain of a `SupHom`. -/
+@[simps]
+protected def withTop (f : SupHom α β) : SupHom (WithTop α) (WithTop β) where
+  toFun := Option.map f
```

Comment:
This should be `WithTop.map` too (and in Lean 3)

Reply:
```suggestion
  toFun := WithTop.map f
```

Reply:
Ah, right!

Reply:
done the latter